wire-ios/Wire-iOS/Sources/UserInterface/Settings/DeviceManagement/RemoveClients/RemoveClientsViewController.swift

```diff
@@ -124,16 +138,24 @@ final class RemoveClientsViewController: UIViewController,
         self.navigationController?.presentingViewController?.dismiss(animated: true, completion: nil)
     }
 
-    func removeUserClient(_ userClient: UserClient) async {
+    func removeUserClient(_ userClient: UserClient) {
+        userClientToDelete = userClient
+        present(requestPasswordController.alertController, animated: true)
+    }
+
+    // MARK: - Helpers
+
+    private func removeUserClient(password: String) async {
+        guard let userClientToDelete = userClientToDelete else { return }
+
         isLoadingViewVisible = true
         do {
-            try await viewModel.removeUserClient(userClient)
-            isLoadingViewVisible = false
+            try await viewModel.removeUserClient(userClientToDelete, password: password)
             delegate?.finishedDeleting(self)
         } catch {
-            isLoadingViewVisible = false
             delegate?.failedToDeleteClients(error)
```

Comment:
Shouldn't we clean the userClientToDelete at some point? I know we teardown everything but just to be safe